src/util/string.cpp

```diff
@@ -3,3 +3,14 @@
 // file COPYING or http://www.opensource.org/licenses/mit-license.php.
 
 #include <util/string.h>
+#include <util/spanparsing.h>
+
+[[nodiscard]] std::vector<std::string> SplitString(const std::string& str, char sep)
```

Comment:
Is the annotation needed in the cpp file as well? At least lock annotations are only supposed to go to the declaration.

Reply:
Good point. Quickly `grep`ing the `[[nodiscard]]` annotation in our codebase shows that there are barely any occurences in .cpp files (only for static functions). Removed it.